plugins/kotlin-dataframe/src/org/jetbrains/kotlinx/dataframe/plugin/impl/api/groupBy.kt

```diff
@@ -199,4 +205,195 @@ abstract class GroupByAggregator(val defaultName: String) : AbstractSchemaModifi
+class GroupByMax0 : GroupByAggregator3(defaultName = "max")
+
+/** Implementation for `std` */
+class GroupByStd0 : GroupByAggregator3(defaultName = "std")
+
+abstract class GroupByAggregator4() : AbstractSchemaModificationInterpreter() {
+    val Arguments.receiver by groupBy()
+
+    override fun Arguments.interpret(): PluginDataFrameSchema {
+        val resolvedColumns = receiver.groups.columns()
+            .filter {
+                it is SimpleDataColumn
+                    && it.type.type.isSubtypeOf(session.builtinTypes.numberType.type, session)
+            }
+
+        return PluginDataFrameSchema(receiver.keys.columns() + resolvedColumns)
+    }
+}
+
+class GroupBySum1 : GroupByAggregator4()
+
+
+
+class GroupByStd1 : GroupByAggregator4()
+
+class GroupByMean1 : GroupByAggregator4()
+
+private fun ConeKotlinType.isSubtypeOfComparable(session: FirSession): Boolean {
+    val comparableTypes: List<FirImplicitBuiltinTypeRef> = listOf(
+        session.builtinTypes.booleanType,
+        session.builtinTypes.numberType,
+        session.builtinTypes.byteType,
+        session.builtinTypes.shortType,
+        session.builtinTypes.intType,
+        session.builtinTypes.longType,
+        session.builtinTypes.doubleType,
+        session.builtinTypes.floatType,
+        session.builtinTypes.uIntType,
+        session.builtinTypes.charType,
+        session.builtinTypes.stringType
+    )
+
+    return comparableTypes.any { it.type.isSubtypeOf(this, session)  }
+}
+
+ /** class FirImplicitThrowableTypeRef(
+    source: KtSourceElement?
+) : FirImplicitBuiltinTypeRef(source, StandardClassIds.Comparable) */
+
+abstract class GroupByAggregatorComparable() : AbstractSchemaModificationInterpreter() {
+    val Arguments.receiver by groupBy()
+
+    override fun Arguments.interpret(): PluginDataFrameSchema {
+        val resolvedColumns = receiver.groups.columns()
+            .filter {
+                it is SimpleDataColumn
+                    && it.type.type.isSubtypeOfComparable(session)
+            }
+
+        return PluginDataFrameSchema(receiver.keys.columns() + resolvedColumns)
+    }
+}
+
+abstract class GroupByAggregatorComparable2(val defaultName: String) : AbstractSchemaModificationInterpreter() {
+    val Arguments.receiver by groupBy()
+    val Arguments.name: String? by arg(defaultValue = Present(null))
+    val Arguments.columns: ColumnsResolver? by arg()
+
+    override fun Arguments.interpret(): PluginDataFrameSchema {
+        if (name == null) { // TODO: add an example, should be double type
+            val resolvedColumns = columns?.resolve(receiver.keys)?.map { it.column }!!.toList()
+            return PluginDataFrameSchema(receiver.keys.columns() + resolvedColumns)
+        } else {
+            val aggregated =
+                makeNullable(
+                    simpleColumnOf(
+                        name ?: defaultName,
+                        session.builtinTypes.doubleType.type
+                    )
```

Comment:
here, I need to create it with type Comparable<Any?>, I don't know is it a good idea or not, but it could be solved only if we updated a sealed hiearchy in Kotlin with `FirImplicitComparableTypeRef` with wrap for class ```
FirImplicitComparableTypeRef(
    source: KtSourceElement?
) : FirImplicitBuiltinTypeRef(source, StandardClassIds.Comparable)
```